### PR TITLE
PS-8626: Inserting JSON data through LOAD DATA INFILE doesn't show fa…

### PR DESCRIPTION
**Pull request by dlenev**
…iled record.

When LOAD DATA or multi-row INSERT statement tried to store bad data in a JSON column an error message was emitted, which didn't reference number of row which caused the problem. Such behavior was not consistent with how similar problems were reported for columns of other types. It didn't allow user to understand which record among inserted was problematic.

This patch solves the problem by changing message for ER_INVALID_JSON_TEXT error, which is emitted in such cases, to mention number of row being inserted. It is easy to do so, as the information about row being processed is already available at the point where error is emitted from Diagnostics_area::current_row_for_condition() member function.

It makes behavior for JSON type consistent with how warnings and errors are reported for other types.

### CODE REVIEW

**Review by percona-ysorokin — APPROVED** [review]
LGTM
The only comment is that we need to merge this directly int o the 8.0.32. So, please, change the base branch of this PR to `release-8.0.32-24` before merging.

**Comment by dlenev** [comment]
Jenkins result doesn't show any failures related to this patch https://ps80.cd.percona.com/view/8.0/job/percona-server-8.0-param-parallel-mtr/45/

**Review by percona-ysorokin — APPROVED** [review]
LGTM (after 8.0.32 rebase)

**Pull request merged**
